### DISCUSSION TITLE
Risks of going below minimum recommended hardware

### DISCUSSION THREAD

**Discussion by strayer** [longhorn/longhorn#3618] [Q&A] [answered]
I'm looking into consolidating the few cloud VMs I have into a low-cost k3s cluster. I currently have one VM with 2 cores and 4 GB memory and two VMs with 1 core and 2 GB memory. The minimum recommended hardware recommends 4 cores and 4 GB of memory.

My cluster won't have anything mission critical on it and especially nothing that really needs a lot of disk I/O. Biggest offenders would most likely be a low-throughput Postgres database and an InfluxDB that gets a bit of data from smart home devices, but also not that much. CPU usage is usually low except some spikes and memory usage would be pretty consistent.

After some tests I think I have to replace the 2 GB memory VMs at least because they would be at 50% memory usage just by running a k3s agent and longhorn without any actual workload. But even then, the cheaper VMs would still only have 2 cores.

All of this would be running on Hetzner Cloud, so I can upgrade to either 3 core / 4 GB memory (8,21 € / month) or 4 core / 8 GB memory (14,76 € / month), but those VMs are quite costly compared to the basic 2 core / 4 GB memory VM (5,83 € / month) I'm using right now.

What I'm wondering now is can I get away with going below the minimum recommended hardware for CPU cores or would I risk getting corrupted data? If too few cores just means maybe degraded performance if suddenly a high I/O load occurs I would probably be fine with that, but corrupted data is obviously a different beast.

**Comment by jenting** [comment, marked as answer]
> What I'm wondering now is can I get away with going below the minimum recommended hardware for CPU cores or would I risk getting corrupted data? If too few cores just means maybe degraded performance if suddenly a high I/O load occurs I would probably be fine with that, but corrupted data is obviously a different beast.

For CPU cores under minimum recommendation, corrupted data probably would not happen.
For Memory and Network under minimum recommendation, it has a chance to happen corrupted data.

However, we did not have a regular test on 2 core / 4 GB combinations.
